pkg/gather/clusterconfig/gather_logs.go

```diff
@@ -0,0 +1,109 @@
+package clusterconfig
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// gatherLogsFromPodsInNamespace collects logs from the pods in provided namespace
+//   - messagesToSearch are the messages to filter the logs(case-insensitive)
+//   - sinceSeconds sets the moment to fetch logs from(current time - sinceSeconds)
+//   - limitBytes sets the maximum amount of logs that can be fetched
+//   - logFileName sets the name of the file to save logs to.
+//   - labelSelector allows you to filter pods by their labels
+// Actual location is `config/pod/{namespace}/logs/{podName}/{fileName}.log`
+func gatherLogsFromPodsInNamespace(
+	ctx context.Context,
+	coreClient v1.CoreV1Interface,
+	namespace string,
+	messagesToSearch []string,
+	sinceSeconds int64,
+	limitBytes int64,
+	logFileName string,
+	labelSelector string,
+) ([]record.Record, error) {
+	pods, err := coreClient.Pods(namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: labelSelector,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var records []record.Record
+
+	for _, pod := range pods.Items {
+		pod, err := coreClient.Pods(namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I guess these 4 lines are not necessary. You already have the pod right?

Reply:
Yup, missed it, thanks :)